src/containers/Header/NotificationsTray/index.jsx

```diff
@@ -34,16 +40,146 @@ class NotificationsTray extends PureComponent {
     this.setState({ viewingNotification: null });
   }
 
+  _onMarkAsReadClick() {
+    if (! this.state.markingMode) {
+      this.setState({
+        markingMode: true,
+      });
+    }
+  }
+
+  _onCancelClick() {
+    this.setState({
+      markingMode: false,
+    });
+  }
+
+  _onMarkClick() {
+    console.log('dega', this.state.notificationIds);
+    this.setState({
+      notificationIds: [],
+      markingMode: false,
+    });
+  }
+
+  _onNotificationCheckBoxClick(checked, notificationId) {
+    if (checked) {
+      const notificationIds = [...this.state.notificationIds, notificationId];
+      this.setState({
+        notificationIds,
+      });
+    } else {
+      const notificationIds = [...this.state.notificationIds.filter(
+        (id) => id !== notificationId
+      )];
+      this.setState({
+        notificationIds,
+      });
+    }
+  }
+
+  _renderNotificationCheckbox(notificationId) {
+    if (this.state.markingMode) {
+      return <Checkbox name={notificationId}
+          checked={this.state.notificationIds.includes(notificationId)}
+          className={styles.NotificationsTray_notification_checkbox}
+          onChange={(checked) => this._onNotificationCheckBoxClick(checked, notificationId)}
```

Comment:
It would be great if we could use these methods directly. `onChange={this._onNotificationCheckBoxClick}`